lib/cache/storages/Storage.ts

```diff
@@ -0,0 +1,6 @@
+export interface Storage {
+  set<V>(key: string, value: V): Storage;
+  get<V>(key: string): V;
+  has(key: string): boolean;
+  delete(key: string): Storage;
+}
```

Comment:
The interface is synchronous, limiting the implementations to only in-memory strategies. The interface definition will not fit an implementation for a Redis or MemCache caches.